warehouse/integrations/deltalake/deltalake.go

```diff
@@ -899,6 +899,9 @@ func (d *Deltalake) partitionQuery(ctx context.Context, tableName string) (strin
 	if !d.config.enablePartitionPruning {
 		return "", nil
 	}
+	if d.Uploader.ShouldOnDedupUseNewRecord() {
+		return "", nil
+	}
```

Comment:
Since the changes are straightforward, going forward with the changes. Also, I have captured the test [here](https://github.com/rudderlabs/rudder-server/pull/3851/files#diff-8ad3c6da0af87741a94ff8d4986b4fb69b117a490a3c8c705e15c97681e43d99R555-R596) since it was easy for me to add it there.